src/ORM/Association.php

```diff
@@ -251,7 +251,7 @@ public function __construct(string $alias, array $options = [])
      */
     public function setName(string $name)
     {
-        if ($this->_targetTable !== null) {
+        if (isset($this->_targetTable)) {
```

Comment:
Why do we need to use cloaking here?

Reply:
This setter itself doesn't make sense to me :slightly_smiling_face: 

- An association will always has a target table.
- Association name is derived from the target table.
- Based on this check and error message the association name should always match the target table alias.
- So what's the point of this setter??

Reply:
Sure, that aside I was just wondering about the !== null change
Is that a PHP81 thing? Due to the typed props?

Reply:
Hmm so here's the thing; association name is set from constructor. The `targetTable` if not specified in constructor option is derived by using the association name as a table alias to get it from the table registry.

So theoretically one can use `setName()` to change association name after object construction until the `targetTable` is derived. But I don't see any practical use case for this.

I suggest simply dropping the setter in 5.0.

Reply:
> Sure, that aside I was just wondering about the !== null change
> Is that a PHP 8 thing? Due to the typed props?

Yes, for typed properties non-initialized vs null are treated differently.

Reply:
This is the PR which add the check for name vs table alias https://github.com/cakephp/cakephp/pull/10161. 
> Association names are mutable, but that change isn't reflected in a collection nor the target table. Changing association name without manually reattaching it in a collection makes it unusable.

So again let's just remove this name setter.

Reply:
> So again let's just remove this name setter.

Yes. This method is/was not a good one.

Reply:
> Yes. This method is/was not a good one.

Okay, I'll deprecate it in 4.next first and then remove here.